src/uri.rs

```diff
@@ -92,14 +92,17 @@ impl Display for RequestUri {
 
 #[test]
 fn test_uri_fromstr() {
```

Comment:
I'd be in favor of this test instead having 2 helpers, to prevent the proliferation of `Some(...)` in every test. Maybe something like:

```rust
assert_eq!(parse("*"), RequestUri::Star);
assert!(parse_err("**"));
```